backend/homework/7/src/main/java/org/example/ServiceVehicle.java

```diff
@@ -0,0 +1,51 @@
+// ServiceVehicle.java
+package org.example;
+
+import jakarta.annotation.PostConstruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+@Component
+public class ServiceVehicle {
+    private static final Logger logger = LoggerFactory.getLogger(ServiceVehicle.class);
+
+    private List<Vehicle> vehicles = new ArrayList<>();
+    private Factory1VehicleService factory1VehicleService;
+    private Factory2VehicleService factory2VehicleService;
+
+    @Autowired
+    public ServiceVehicle(Factory1VehicleService factory1VehicleService, Factory2VehicleService factory2VehicleService) {
+        this.factory1VehicleService = factory1VehicleService;
+        this.factory2VehicleService = factory2VehicleService;
+    }
+
+    public Vehicle findMostExpensiveVehicle() {
+        return vehicles.stream()
```

Comment:
it is possible that the object will be null, so it's important to provide a proper message or handling mechanism for that scenario.